### ISSUE TITLE
Steam crashes with locale ru_UA.utf8

### ISSUE THREAD

**Issue by Shished** [ValveSoftware/steam-for-linux#3897]
```
Steam: An X Error occurred
X Error of failed request:  BadValue (integer parameter out of range for operation)
Major opcode of failed request:  18 (X_ChangeProperty)
Value in failed request:  0xffffffff
Serial number of failed request:  101
```

```
locale
LANG=ru_UA.utf8
LC_CTYPE="ru_UA.utf8"
LC_NUMERIC="ru_UA.utf8"
LC_TIME="ru_UA.utf8"
LC_COLLATE="ru_UA.utf8"
LC_MONETARY="ru_UA.utf8"
LC_MESSAGES="ru_UA.utf8"
LC_PAPER="ru_UA.utf8"
LC_NAME="ru_UA.utf8"
LC_ADDRESS="ru_UA.utf8"
LC_TELEPHONE="ru_UA.utf8"
LC_MEASUREMENT="ru_UA.utf8"
LC_IDENTIFICATION="ru_UA.utf8"
LC_ALL=
```

I launched locale test from #1420 
```
#include <stdio.h>
#include <clocale>
#include <X11/Xlib.h>
#include <X11/Xlocale.h>

int main(int argc, char **argv) {
setlocale(LC_CTYPE, "ru_UA.utf8");
if (!XSupportsLocale())
printf("locale not supported.\n");
else
printf("locale supported.\n");
return 0;
}
```
It says "locale not supported."

Steam works if LANG or LC_ALL is set to C or en_US.utf8. This fixes the issue.

PC Specs from Steam (translated): http://pastebin.com/FWqgTyYA

**Comment by Tele42** [comment]
Does "ru_UA.UTF-8" work? http://unix.stackexchange.com/questions/186963/what-is-the-proper-encoding-name-to-use-in-locale-for-utf-8 suggests that is what X is looking for.

**Comment by Shished** [comment]
export LANG=ru_UA.UTF-8 changes value for all locale related variables.

Tester app says that this locale is supported and Steam starts without problems.

How should i fix this problem? Some system config should be changed.

**Comment by gdrewb-valve** [comment]
Keeping open as Steam shouldn't crash but this doesn't appear to be a Steam problem at the root.  Is there a reason you need to use .utf8 instead of the correct .UTF-8?

**Comment by Shished** [comment]
I don't think that any app would notice the change, especially when the value was incorrect.

I changed my locale to ru_UA.UTF-8 and that fixed my problem.

**Comment by Tele42** [comment]
@gdrewb-valve There are more glibc accepted locales than Xorg, we have seen this issue before, such as #3027, some locales have .utf8 -> .UTF-8 aliases (on Xorg's side) while others do not.

**Comment by gdrewb-valve** [comment]
Interesting, thanks for the info.

**Comment by Tak** [comment]
Similar crash if locale is set to en_AG (which can apparently happen if you select "English" with no country specializer in some gui settings tools).
Workaround is easy; discovering the source of the problem is difficult.

**Comment by Tids** [comment]
also happens for `ast_ES.UTF-8`
